File: 1 Python/Leetcode-2220-MinBitFlipsToConvertNumber.py
# given the constraints we operate in 32bit representation to find flips
class Solution:
    def minBitFlips(self, start: int, goal: int) -> int:
        s32b = format(start,'032b')[2:]
        g32b = format(goal,'032b')[2:]
        moves = 0
        for c in range(len(g32b)):
            if s32b[c]!=g32b[c]:
                moves+=1
        return moves

File: 2 CPP/Leetcode-2220-MinBitFlipsToConvertNumber.cpp
//0ms Solution
//
//
class Solution {
public:
    int minBitFlips(int start, int goal) {
        string s32 = bitset<32>(start).to_string();
        string g32 = bitset<32>(goal).to_string();
        int moves = 0;
        for(int i=0;i<g32.size();++i){
            moves+= (g32[i]!=s32[i]); //boolean addition
        }
        return moves;
    }
};
